api/scanner/exif/exif_parser_internal.go

```diff
@@ -142,8 +142,16 @@ func (p internalExifParser) ParseExif(media_path string) (returnExif *models.Med
 
 	lat, long, err := exifTags.LatLong()
 	if err == nil {
-		newExif.GPSLatitude = &lat
-		newExif.GPSLongitude = &long
+		// **Added GPS data validation**
+		// https://github.com/photoview/photoview/issues/850
+		if lat < -90 || lat > 90 || long < -90 || long > 90 {
+			returnErr = errors.New(fmt.Sprintf("Incorrect GPS data in the %s Exif data: %f, %f, while expected values between '-90' and '90'", media_path, long, lat))
```

Comment:
- No comment
- I don't believe the nil setting is required on this based on the context I can see, we should be able to invert the condition and if it's within the correct range apply it.

My same comment re: `math.Abs()` applies here too.